learning_observer/learning_observer/learning_tools/canvas.py

```diff
@@ -0,0 +1,162 @@
+        
+        # Handle invalid/expired token
+        if response.status_code == 401 and retry:
+            new_tokens = self.refresh_tokens()
+            if 'access_token' in new_tokens:
+                self.update_access_tokens(new_tokens['access_token'])
+                return self.api_call(method, endpoint, params, data, absolute_url, retry=False)
+        
+        if response.status_code != 200:
+            response.raise_for_status()
+        
+        return response.json()
+
+    def refresh_tokens(self):
+        url = f'https://{self.defaultServer}/login/oauth2/token'
+        params = {
+            "grant_type": "refresh_token",
+            "client_id": self.client_id,
+            "client_secret": self.client_secret,
+            "refresh_token": self.refresh_token
+        }
+        return self.api_call('POST', url, params=params, absolute_url=True)
+
+    def get_account(self):
+        return self.api_call('GET', '/accounts/self')
+    
+    def get_users(self):
+        return self.api_call('GET', '/accounts/self/users')
+
+    def get_courses(self):
+        return self.api_call('GET', '/courses', params={'per_page': self.defaultPerPage})
+    
+    def get_course_enrollment(self, course_id):
+        return self.api_call('GET', f'/courses/{course_id}/students', params={'per_page': self.defaultPerPage})
+    
+    def get_course_teachers(self, course_id):
+        return self.api_call('GET', f'/courses/{course_id}/users', params={'per_page': self.defaultPerPage, 'enrollment_type[]': 'teacher'})
+
+    def get_course_assignments(self, course_id):
+        return self.api_call('GET', f'/courses/{course_id}/assignments', params={'per_page': self.defaultPerPage})
+    
+    def create_user(self, account_id, user_data):
+        endpoint = f'/accounts/{account_id}/users'
+        data = {
+            'user': {
+                'name': user_data['name'],
+                'sortable_name': user_data.get('sortable_name', user_data['name']),
+                'terms_of_use': True
+            },
+            'pseudonym': {
+                'unique_id': user_data['email'],
+                'password': user_data['password'],
+                'sis_user_id': user_data.get('sis_user_id'),
+                'integration_id': user_data.get('integration_id')
+            },
+            'communication_channel': {
+                'type': 'email',
+                'address': user_data['email'],
+                'skip_confirmation': True
+            }
+        }
+        return self.api_call('POST', endpoint, data=data)
+    
+    def enroll_user(self, course_id, user_id, enrollment_type='StudentEnrollment'):
+        endpoint = f'/courses/{course_id}/enrollments'
+        data = {
+            'enrollment': {
+                'user_id': user_id,
+                'type': enrollment_type,
+                'enrollment_state': 'active'
+            }
+        }
+        return self.api_call('POST', endpoint, data=data)
+
+def test():
```

Comment:
How could someone go about testing this? Is it as easy as just running this method? It might need a docstring then.